File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    Set<Integer> primes = new HashSet<>();
    
    public boolean isPrime(int num) {
        if (num == 0 || num == 1) return false;
        for (int i = 2; i < num / 2 + 1; i++) {
            if (num % i == 0) return false;
        }
        return true;
    }
    
    public void backtrack(String temp, List<Boolean> used, List<String> nums) {
        if (temp.length() != 0 && isPrime(Integer.parseInt(temp))) {
            primes.add(Integer.parseInt(temp));
        }
        for (int i = 0; i < nums.size(); i++) {
            if (!used.get(i)) {
                used.set(i, true);
                backtrack(temp + nums.get(i), used, nums);
                used.set(i, false);
            }
        }
    }
    
    public int solution(String numbers) {
        List<String> nums = new ArrayList<>();
        List<Boolean> used = new ArrayList<>();
        for(int i = 0; i < numbers.length(); i++) {
            nums.add(Character.toString(numbers.charAt(i)));
            used.add(false);
        }
        backtrack("", used, nums);
        return primes.size();
    }
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def solution(numbers):
    
    def is_prime(num):
        if num == 0 or num == 1:
            return False
        
        for i in range(2, num // 2 + 1):
            if num % i == 0:
                return False
        return True
    
    primes = set()
    nums = []
    for n in numbers:
        nums.append(n)
        
    for r in range(1, len(nums) + 1):
        for p in permutations(nums, r):
            num = int(''.join(map(str, p)))
            if is_prime(num):
                primes.add(num)
                
    return len(primes)